Fix issues

erikbrudzinskis opened the pull request:
### Fixes #4 
- The output now has correct branch and version.

### Closes #7 
- Adds an additional parameter "checkForUncommittedChanges". If set to true and there are uncommitted changes, then aborts the procedure.

### Closes #9
- Allows to update any old version to SNAPSHOT.

_Not entirely sure what's happened there. versions-maven-plugin:set does not seem to assume anything about the branches. It looks like it was trying to update the wrong version (i.e. 0.0.5 -> 0.0.7-SNAPSHOT, when it should have been 0.0.6 -> 0.0.7-SNAPSHOT). It couldn't find it and ignored "set version" altogether. As a result, it didn't work for me on any branch._

_Possibly, there was a race condition, and sometimes it worked, sometimes it didn't. I.e. if MavenProject is correctly updated with the correct release version before we attempt to change it to snapshot, then it would work. Or it could have been some changes on Git side?_

_Anyway, allowing to update any old version seems to do the trick. Tested locally with main and master branches on different projects._

### Other
- Java version is now updated to 11.
- Dependencies now updated to avoid high-risk vulnerabilities.

anton-johansson commented on src/main/java/com/antonjohansson/versionlifecycleplugin/ValidateMojo.java:
```diff
@@ -0,0 +1,24 @@
+public class ValidateMojo extends AbstractVersionMojo
+{
+    @Parameter(defaultValue = "${checkForUncommittedChanges}")
+    boolean checkForUncommittedChanges = true;
```
Change this to `private`.

erikbrudzinskis replied:
Changed!

anton-johansson commented on src/main/java/com/antonjohansson/versionlifecycleplugin/ValidateMojo.java:
```diff
@@ -0,0 +1,24 @@
+@Mojo(name = "validate")
+public class ValidateMojo extends AbstractVersionMojo
+{
+    @Parameter(defaultValue = "${checkForUncommittedChanges}")
```
Here's how we've done for other parameters:
```java
@Parameter(name = "releaseCommitMessagePattern", property = "version.releaseCommitMessagePattern", defaultValue = "[version]")
```

So I suggest:

```diff
-    @Parameter(defaultValue = "${checkForUncommittedChanges}")
+    @Parameter(name = "checkForUncommittedChanges", property = "version.checkForUncommittedChanges", defaultValue = "true")
```

erikbrudzinskis replied:
Changed!

anton-johansson reviewed (changes requested):
Nice work! Some small things, otherwise mergeable. How should we handle the commits? Squash everything into one big commit that closes the three issues?

anton-johansson commented:
Hmm, the `patch` command no longer works.

anton-johansson commented:
I stole your commit that fixes the unit tests and put it in this one instead: #13. Lets get that one merged first, so we get the CI build working.

erikbrudzinskis commented:
I have added the changes to patch - to not create snapshots. And squashed all commits.

You mentioned that `patch` does not work - it seems to work fine for me, could you give some more info so I can replicate?
